utils/telemetry/service.go

```diff
@@ -31,17 +32,25 @@ func (s *Service) SetUser(userID string) {
 	s.userID = userID
 }
 
+// SetVersion sets the realm-cli version for this service
+func (s *Service) SetVersion(version string) {
+	s.version = version
+}
+
 // TrackEvent tracks the event based on the tracker
 func (s *Service) TrackEvent(eventType EventType, data ...EventData) {
-	s.tracker.Track(event{
-		id:          primitive.NewObjectID().Hex(),
-		eventType:   eventType,
-		userID:      s.userID,
-		time:        time.Now(),
-		executionID: s.executionID,
-		command:     s.command,
-		data:        data,
-	})
+	if s.tracker != nil {
```

Comment:
instead of the `nil` check here, let's just initialize the telemetry service as `&telemetry.Service{Tracker: &noopTracker{}}` in `main.go`

then in `Setup`, you can also remove the `noopTracker` assignment since it will already be there